src/linter.coffee

```diff
@@ -1,38 +1,44 @@
-_ = require 'lodash'
-async = require 'async'
-asyncHandlers = require 'async-handlers'
-fsExtra = require 'fs-extra'
-DependencyLinter = require './linter/dependency_linter'
-UsedModuleFinder = require './linter/used_module_finder'
-path = require 'path'
-
-
 class Linter
 
 
-  constructor: (@dir, {allowUnused, devFilePatterns, devScripts, ignoreFilePatterns}) ->
-    @dependencyLinter = new DependencyLinter {allowUnused, devFilePatterns, devScripts}
-    @usedModuleFinder = new UsedModuleFinder {@dir, ignoreFilePatterns}
+  constructor: (@dir, {@allowUnused, @devFilePatterns, @devScripts, @ignoreFilePatterns}) ->
 
 
-  lint: (done) ->
-    async.parallel {
-      listedModules: @getListedModules
-      usedModules: @usedModuleFinder.find
-    }, asyncHandlers.transform(@dependencyLinter.lint, done)
+  lint: (done) =>
+    async = require 'async'
+    asyncHandlers = require 'async-handlers'
```

Comment:
The first point is something I like about it. The complexity is the same to me as you are still using the same variables its just moving where something is defined.

No rebuttal for the second point that is a downside to this approach

To the third point, I'm leaning towards moving anything that is required multiple times to an outer scope so that it can stay dry